chemprop/train/molecule_fingerprint.py

```diff
@@ -0,0 +1,163 @@
+        if all(mol is not None for mol in full_data[full_index].mol):
+            full_to_valid_indices[full_index] = valid_index
+            valid_index += 1
+
+    test_data = MoleculeDataset([full_data[i] for i in sorted(full_to_valid_indices.keys())])
+
+    # Edge case if empty list of smiles is provided
+    if len(test_data) == 0:
+        return [None] * len(full_data)
+
+    print(f'Test size = {len(test_data):,}')
+
+    # Create data loader
+    test_data_loader = MoleculeDataLoader(
+        dataset=test_data,
+        batch_size=args.batch_size,
+        num_workers=args.num_workers
+    )
+
+    print(f'Encoding smiles into a fingerprint vector from a single model')
+    assert len(args.checkpoint_paths) == 1, "Fingerprint generation only supports one model, cannot use an ensemble"
+    # Load model
+    model = load_checkpoint(args.checkpoint_paths[0], device=args.device)
+
+    model_preds = model_fingerprint(
+        model=model,
+        data_loader=test_data_loader
+    )
+
+    # Save predictions
+    print(f'Saving predictions to {args.preds_path}')
+    assert len(test_data) == len(model_preds)
+    makedirs(args.preds_path, isfile=True)
+
+    # Copy predictions over to full_data
+    for full_index, datapoint in enumerate(full_data):
+        valid_index = full_to_valid_indices.get(full_index, None)
+        preds = model_preds[valid_index] if valid_index is not None else ['Invalid SMILES'] * (args.hidden_size*args.number_of_molecules)
+
+        fingerprint_columns=[f'fp_{i}' for i in range(args.hidden_size*args.number_of_molecules)]
+        for i in range(len(fingerprint_columns)):
+            datapoint.row[fingerprint_columns[i]] = preds[i]
+
+    # Write predictions
+    with open(args.preds_path, 'w') as f:
+        # writer = csv.DictWriter(f, fieldnames=full_data[0].row.keys())
+        writer = csv.DictWriter(f, fieldnames=args.smiles_columns+fingerprint_columns,extrasaction='ignore')
+        writer.writeheader()
+        # writer = csv.writer(f)
+
+        # header = []
+
+        # header.extend(['smiles'])
+        # header.extend(['fp{}'.format(x) for x in range(1,args.hidden_size*args.number_of_molecules+1)])
+
+        # writer.writerow(header)
+
+        for datapoint in full_data:
+            writer.writerow(datapoint.row)
+            # row = []
+
+            # row.extend(datapoint.smiles)
+
+            # if model_preds[i] is not None:
+            #     row.extend(model_preds[i][:args.hidden_size*args.number_of_molecules])
+            # else:
+            #     row.extend(['Invalid input'] * args.hidden_size*args.number_of_molecules)
+
+            # writer.writerow(row)
+
+    return model_preds
+
+def model_fingerprint(model: MoleculeModel,
+            data_loader: MoleculeDataLoader,
+            disable_progress_bar: bool = False) -> List[List[float]]:
+    """
+    Encodes the provided molecules into the latent fingerprint vectors, according to the provided model.
+
```

Comment:
I believe lines 133-135 should read (typo on `data_loader`, added `disable_progress_bar` description, deleted `batch_size description`):

```
:param model: A :class:`~chemprop.models.model.MoleculeModel`.
:param data_loader: A :class:`~chemprop.data.data.MoleculeDataLoader`.
:param disable_progress_bar: Whether to disable the progress bar.
```